lms/djangoapps/bookmarks/services.py

```diff
@@ -31,7 +59,13 @@ def bookmarks(self, course_key):
         Returns:
             list of dict:
         """
-        return api.get_bookmarks(self._user, course_key=course_key, fields=DEFAULT_FIELDS + OPTIONAL_FIELDS)
+        bookmarks = api.get_bookmarks(self._user, course_key=course_key, fields=DEFAULT_FIELDS + OPTIONAL_FIELDS)
+
+        if course_key.run is not None:
```

Comment:
I got this issue in tests that in some cases the run was missing, the `course_key` could not be serialized and so the `cache_key` could not be constructed. Is there a way to work around that? I don't suppose this will be an issue for a valid course?